packages/axios-extension/src/abap/adt/handlers/transport.ts

```diff
@@ -0,0 +1,97 @@
+import XmlParser from 'fast-xml-parser';
+import * as xpath from 'xpath';
+import { DOMParser } from '@xmldom/xmldom';
+
+export function parseTransportRequests(xml: string): string[] {
+    if (XmlParser.validate(xml) !== true) {
+        return [];
+    }
+    const doc = new DOMParser().parseFromString(xml);
+    return getTransportChecksResponse(doc);
+}
+
+const LocalPackageText = ['LOCAL_PACKAGE', 'LOCAL'];
+const enum AdtTransportStatus {
+    Success = 'S',
+    Error = 'E'
+}
+
+/**
+ * Implementation based on WebIDE with simplified error handling.
+ * Might need to improve the error handling in the future to expose
+ * the backend error messages to end user / API consumer in the future.
+ *
+ * @param doc
+ * @returns available transport numbers
+ */
+function getTransportChecksResponse(doc: Document): string[] {
+    const status = xpath.select1('//RESULT/text()', doc)?.toString();
+
+    switch (status) {
+        case AdtTransportStatus.Success:
+            return getTransportList(doc);
+        case AdtTransportStatus.Error:
+        default:
+            return [];
+    }
+}
+
+/**
+ * Provide a list of transport numbers available for the input package name and project name
+ * in a ADT CTS request.
+ *
+ * @param doc
+ * @returns
+ * - For local package, an array list that contain a single empty string is returned.
+ * - For errors or other unkonwn reasons no transport number found, an empty array list is returned.
+ */
+function getTransportList(doc: Document): string[] {
+    const recording = xpath.select1('//RECORDING/text()', doc)?.toString();
+    const locked = (xpath.select1('//LOCKS', doc) as Element)?.textContent;
+    const localPackage = xpath.select1('//DLVUNIT/text()', doc)?.toString();
```

Comment:
Are these paths part of some schema? Do we need to consider versions if such a schema is available? Can we log anything as an early warning if the schema changes underfoot?

Reply:
Based on the WebIDE implementation, there is no need to handle schema changes at the response data level.